src/languageservice/parser/yaml-documents.ts

```diff
@@ -105,6 +105,10 @@ export class SingleYAMLDocument extends JSONDocument {
       return [this.findClosestNode(positionOffset, textBuffer, configuredIndentation), true];
     }
 
+    const textAfterPosition = lineContent.substring(position.character);
+    const spacesAfterPositionMatch = textAfterPosition.match(/^([ ]+)\n?$/);
```

Comment:
@p-spacek can you add a comment briefly explaining what this regex does, eg '// regex finds all spaces and newlines`